### PR TITLE
Change blob storage to utilize all data.paths

### PR DESCRIPTION
**Pull request by mfussenegger**
- BlobIndex/BlobShard are no longer created using injection but instead
  the BlobIndicesService uses a LifecycleListener and manages the
  instances itself.

- ShardCollectService is no longer a eager singleton because it depends
  on BlobShard which is now created after the shard-injector has been
  created.

### CODE REVIEW

**Comment by mfussenegger** [comment]
@dobe please review

File: blob/src/main/java/io/crate/blob/v2/BlobIndicesService.java [hunk 1]
```diff
@@ -130,6 +131,34 @@ public BlobShard blobShardSafe(ShardId shardId) {
         return blobShardSafe(shardId.getIndex(), shardId.id());
     }
 
+    private class LifecycleListener extends IndicesLifecycle.Listener {
+
+        @Override
+        public void afterIndexCreated(IndexService indexService) {
+            indices.put(indexService.index().getName(), new BlobIndex(globalBlobPath));
```
Comment (dobe) [inline blob/src/main/java/io/crate/blob/v2/BlobIndicesService.java]: pls assert that it is not already there

Comment (dobe) [inline blob/src/main/java/io/crate/blob/v2/BlobIndicesService.java]: check if blobs are enabled on this index, also for any other listener methods

Comment (dobe) [inline blob/src/main/java/io/crate/blob/v2/BlobIndicesService.java]: pls assert in a test that no normal indices/shards are in the service

Comment (mfussenegger) [inline blob/src/main/java/io/crate/blob/v2/BlobIndicesService.java]: There is an `@After` assertion to make sure no indices are leaked: https://github.com/crate/crate/pull/4482/files#diff-97b4abd1024416837a21d1f1d8c1a106R58

**Comment by mfussenegger** [comment]
That's our current per-table blob_path documentation https://github.com/crate/crate/blob/master/blackbox/docs/blob.txt#L52 ?

**Pull request merged**